platform/PlatformTasks.roc

```diff
@@ -47,12 +49,28 @@ hosted PlatformTasks
         InternalPath,
     ]
 
+InternalIOErr : {
+    tag : [
+        BrokenPipe,
+        WouldBlock,
+        WriteZero,
+        Unsupported,
+        Interrupted,
+        OutOfMemory,
+        UnexpectedEof,
+        InvalidInput,
+        Other,
+    ],
+    msg : Str,
+}
+
 stdoutLine : Str -> Task {} Str
 stdoutWrite : Str -> Task {} Str
 stderrLine : Str -> Task {} Str
 stderrWrite : Str -> Task {} Str
-stdinLine :  Task Str Str
+stdinLine : Task Str Str
 stdinBytes : Task (List U8) {}
+stdinReadToEnd : Task (List U8) InternalIOErr
```

Comment:
We'd previously removed all the glue from this platform. Now we are more confident in how to use it (with more manual oversight), I have added it back in here to improve the quality of error reporting. We can use this `InternalIOErr` to support the errors for all the stdio effects. But I'll leave that for a future PR.